crates/ruff_linter/src/checkers/ast/analyze/statement.rs

```diff
@@ -154,6 +154,17 @@ pub(crate) fn statement(stmt: &Stmt, checker: &mut Checker) {
                     parameters,
                 );
             }
+            if checker.enabled(Rule::GeneratorReturnFromIterMethod) {
+                flake8_pyi::rules::bad_generator_return_type(
+                    checker,
+                    stmt,
+                    *is_async,
+                    name,
+                    returns.as_ref().map(AsRef::as_ref),
+                    parameters,
+                    body,
```

Comment:
This is really common in our existing rules, but going forward, the preferred pattern here is to pass in `function_def` and access the fields within `bad_generator_return_type`, rather than passing in each field individually. (In other words: would recommend removing all these arguments, and just calling in `function_def`; you probably don't even need `stmt`).

The benefit is: the signature is much clearer, and we prevent callers from passing in bad data by accident (e.g., `body` here would accept _any_ list of statements, not _just_ a function body).

Reply:
Ahh thanks! I was using https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/flake8_pyi/rules/non_self_return_type.rs as my model for this PR, as it seemed to be a fairly similar rule

Reply:
:thumbsup: Makes total sense. All of those rule signatures should be updated in theory but in practice it hasn’t merited doing, so more just migrating them as we go.